prereise/gather/demanddata/nrel_efs/tests/test_get_efs_data.py

```diff
@@ -1,107 +1,138 @@
+    "Flexibility": ["Base"] * 4 * 48 * 8760,
+    "Year": [2030] * 4 * 48 * 8760,
+    "LocalHourID": sorted([i for i in range(1, 8761)] * 4 * 48),
+    "State": sorted([i for i in cont_states] * 4) * 8760,
+    "Sector": ["Commercial", "Industrial", "Residential", "Transportation"] * 48 * 8760,
+    "LoadMW": [1, 2, 3, 4] * 48 * 8760,
+}
+dummy_flex_df = pd.DataFrame(data=dummy_flex_data)
+dummy_flex_df.to_csv("EFSFlexLoadProfiles_High.csv", index=False)
+
+
+def test_check_electrification_scenarios_for_download():
+    # Run the check
+    test_es = _check_electrification_scenarios_for_download(es={"All"})
+
+    # Specify the expected values
+    exp_es = {"Reference", "Medium", "High"}
+
+    # Compare the two sets
+    assert test_es == exp_es
+
+
+def test_check_technology_advancements_for_download():
+    # Run the check
+    test_ta = _check_technology_advancements_for_download(ta={"All"})
+
+    # Specify the expected values
+    exp_ta = {"Slow", "Moderate", "Rapid"}
+
+    # Compare the two sets
+    assert test_ta == exp_ta
+
+
+def test_check_path():
+    # Run check
+    test_fpath = _check_path(fpath="")
+
+    # Specify the expected file path
+    exp_fpath = os.getcwd()
+
+    # Compare the two file paths
+    assert test_fpath == exp_fpath
+
+
+def test_download_data():
+    # Download a file using _download_data
+    _download_data(
+        zip_name="project_resstock_efs_2013.zip",
+        url="https://data.nrel.gov/system/files/128/project_resstock_efs_2013.zip",
+        fpath="",
+    )
 
-def test_download_demand_data():
-    # Download one of the EFS demand data sets
-    download_demand_data(es={"Reference"}, ta={"Slow"})
-
-    try:
-        # Load the downloaded EFS demand data set
-        df = pd.read_csv("EFSLoadProfile_Reference_Slow.csv")
-
-        # Access the columns
-        test_cols = list(df.columns)
-        exp_cols = [
-            "Electrification",
-            "TechnologyAdvancement",
-            "Year",
-            "LocalHourID",
-            "State",
-            "Sector",
-            "Subsector",
-            "LoadMW",
-        ]
-
-        # Compare the two values
-        assert len(test_cols) == len(exp_cols)
-
-        # Remove the downloaded EFS data set
-        os.remove("EFSLoadProfile_Reference_Slow.csv")
-
-    except FileNotFoundError:
-        # If the automated extraction did not work, check that the .zip file was created
-        assert os.path.isfile("EFSLoadProfile_Reference_Slow.zip")
-
-        # Remove the downloaded .zip file
-        os.remove("EFSLoadProfile_Reference_Slow.zip")
-
+    # Check that the expected .zip file was downloaded
+    assert os.path.isfile("project_resstock_efs_2013.zip")
 
-def test_download_flexibility_data():
-    # Download one of the EFS flexibility data sets
-    download_flexibility_data(es={"Reference"})
+    # Remove the downloaded .zip file
+    os.remove("project_resstock_efs_2013.zip")
```

Comment:
This should be the `finally` block corresponding to a `try` block around the `_download_data` and `assert` calls, so that the cleanup happens no matter what.

Reply:
Ah, thanks for this tip! I'll make that change.

Reply:
Done.